fix: switch send_email to use boto3

pwnage101 opened the pull request:
```
commit 96d5f46ade46f49a2a725e7b5bbb29e6b088cdd9
Author: Troy Sankey <tsankey@edx.org>
Date:   Fri Apr 16 17:40:09 2021 -0400

    chore: update CI to test python 3.7, 3.8, remove 3.5
    
    Python 3.5 was *already* not compatible with this repo since there were
    several packages in requirements/base.txt that were not even installable
    with 3.5.  Also, I specifically opted to skip python 3.6 because even
    Ubuntu oldstable (18.04) ships python 3.8 via standard bionic
    repositories.

commit cd3c13ac81b3a44a33fbe3d1776ae749a396d1e9
Author: Troy Sankey <tsankey@edx.org>
Date:   Fri Apr 16 16:36:08 2021 -0400

    fix: switch send_email to use boto3
    
    boto2 send_email() outright doesn't work anymore, and the package is
    totally out of maintenance now so our only option is to migrate to
    boto3.  This was brought to our attention when it started breaking the
    user retirement pipeline (the hubspot stage which sends a notification
    email internally).  The AWS error message was clear that we must upgrade
    our API client (boto2).
    
    I also ran make upgrade.

commit 1552c897ad6d6423bd50bd1053fb95ce11b7d303
Author: Troy Sankey <tsankey@edx.org>
Date:   Fri Apr 16 16:32:32 2021 -0400

    fix: remove --no-index from pip-compile call, and run make upgrade
    
    This option doesn't exist, plain and simple.  I'm not sure how it made
    it's way into the Makefile to begin with.  It breaks make upgrade, so it
    has to go!
```

bmedx commented:
Seems ok on the surface, but I don't remember if we're running 3.5 in production for these jobs or not. Either way I guess we need to yank test support for it (at least in Travis).

pwnage101 commented:
> Seems ok on the surface, but I don't remember if we're running 3.5 in production for these jobs or not. Either way I guess we need to yank test support for it (at least in Travis).

Yanked 3.5, added 3.7 and 3.9 (see 3rd commit I just added).

pwnage101 commented:
Okay, just kidding about python 3.9.  `backoff`, among others, doesn't support 3.9 yet!  I'll "back off" a little.

pwnage101 commented:
For the record, the user retirement pipeline currently runs python 3.8.

jazibhumayun commented on requirements/base.in:
```diff
@@ -3,6 +3,7 @@
 atlassian-python-api
 backoff==1.5.0
 boto==2.43.0
```
Is this boto version correct here? @pwnage101

jazibhumayun replied:
I see you have added boto3 as well but kept it the same on this line, just highlighting it to confirm if this is intentional ?

pwnage101 replied:
Yes, there's a lot of other code that still uses boto 2 and I didn't intent to change the behavior of any of those code paths.  I only want to update the call to send_email to use boto3

jazibhumayun replied:
Ok alright

jazibhumayun replied:
Should we merge this PR now or wait this Wednesday?  @pwnage101

jazibhumayun reviewed (approved):
LGTM (y)
